### PR TITLE
ENG-3941 Token and API payload changes

### PR DESCRIPTION
**Pull request by zonia3000**

### CODE REVIEW

**Review by cecchisandrone — APPROVED** [review]

File: microservices/app-builder-menu-bff/src/client/requestConfig.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    const requestConfig: AxiosRequestConfig<any> = {}
+
+    // Forward Authorization header
+    const authHeader = req.header('Authorization')
+    if (authHeader !== undefined) {
+        requestConfig.headers = { Authorization: authHeader }
+    }
+
+    return requestConfig
+}
```
Comment (Dv92) [inline microservices/app-builder-menu-bff/src/client/requestConfig.ts]: if you want, here you can prevent the `any` inference type in this way:

```javascript
import { Request } from 'express';
import { AxiosRequestConfig, AxiosRequestHeaders } from 'axios';

type RequestConfig = AxiosRequestConfig<{
    headers?: AxiosRequestHeaders
}>

export function getRequestConfig(req: Request) {
  const requestConfig: RequestConfig = {};

  // Forward Authorization header
  const authHeader = req.header('Authorization');
  if (authHeader !== undefined) {
    requestConfig.headers = { Authorization: authHeader };
  }

  return requestConfig;
}
```

In this way, when you inspect the function return you have a predictable type of config that is used.

**Review by Dv92 — APPROVED** [review]

File: microservices/app-builder-menu-bff/src/client/requestConfig.ts (hunk 1 above)
Comment (zonia3000) [reply microservices/app-builder-menu-bff/src/client/requestConfig.ts]: Done, thanks

File: microservices/app-builder-menu-bff/src/client/response/IBundleWidget.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  }
+  labels?: {
+    pbcNames?: string[]
+  }
+}
+
+export type IBundle = {
+  bundleId: string
+  bundleName: string
+  bundleCode?: string // TODO: remove '?' when this will be populated by API
```
Comment (cecchisandrone) [inline microservices/app-builder-menu-bff/src/client/response/IBundleWidget.ts]: I think we can remove this now?

File: microservices/app-builder-menu-bff/src/service/listNav.ts [hunk 3]
```diff
@@ -1,25 +1,71 @@
...
+  pbcName: string
+  bundleCode: string
+  mfeName: string | null // mfeName is null for global nav
+  label: { [key: string]: string }
+  target: string
+  addr: string
+  organization: string
+}
+
+const LOCAL_PBC_NAME = ':local'
```
Comment (cecchisandrone) [inline microservices/app-builder-menu-bff/src/service/listNav.ts]: I would use `Uncategorized` as label

File: microservices/app-builder-menu-bff/src/client/response/IBundleWidget.ts (hunk 2 above)
Comment (zonia3000) [reply microservices/app-builder-menu-bff/src/client/response/IBundleWidget.ts]: Endpoint `/bundles` at the moment is not returning `bundleCode`, `descriptorExt` and `labels`.

**Pull request merged**